From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 14 Oct 2020 21:17:13 +0200
Subject: seat-native: Fix has_touchscreen becoming FALSE when a non
 touchscreen device gets added

Unconditionally setting has_touchscreen to check_touch_mode
when a new device gets added leads to has_touchscreen becoming
false when during runtime e.g. an USB keyboard gets plugged in.

Fix this by setting has_touchscreen to TRUE when check_touch_mode
is TRUE and leaving it alone otherwise.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1506
(cherry picked from commit 6c240dc83b33d26696825c6ee70560e201542fd7)

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1610>

Origin: upstream, 3.38.2, commit:48d4153c06979db1b21e82dc34d801c32ac80b9a
---
 src/backends/native/meta-seat-native.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/backends/native/meta-seat-native.c b/src/backends/native/meta-seat-native.c
index 5fb0e8e..5c62805 100644
--- a/src/backends/native/meta-seat-native.c
+++ b/src/backends/native/meta-seat-native.c
@@ -1450,7 +1450,8 @@ meta_seat_native_handle_device_event (ClutterSeat  *seat,
   switch (event->type)
     {
       case CLUTTER_DEVICE_ADDED:
-        seat_native->has_touchscreen = check_touch_mode;
+        if (check_touch_mode)
+          seat_native->has_touchscreen = TRUE;
 
         if (libinput_device_has_capability (device_native->libinput_device,
                                             LIBINPUT_DEVICE_CAP_SWITCH) &&
